### PR TITLE
Fix pull empty image URL

### LINKED ISSUE #98628: Fail pre-pulling images
**What happened**:
```
I0131 19:16:54.969991   29670 e2e_node_suite_test.go:185] Pre-pulling images so that they are cached for the tests.
I0131 19:16:54.980193   29670 image_list.go:158] Pre-pulling images with docker [ docker.io/library/busybox:1.29 docker.io/library/httpd:2.4.38-alpine docker.io/library/nginx:1.14-alpine docker.io/library/perl:5.26 docker.io/nfvpe/sriov-device-plugin:v3.1 gcr.io/kubernetes-e2e-test-images/ipc-utils:1.0 gcr.io/kubernetes-e2e-test-images/node-perf/npb-ep:1.0 gcr.io/kubernetes-e2e-test-images/node-perf/npb-is:1.0 gcr.io/kubernetes-e2e-test-images/node-perf/tf-wide-deep-amd64:1.0 gcr.io/kubernetes-e2e-test-images/nonewprivs:1.0 gcr.io/kubernetes-e2e-test-images/nonroot:1.0 gcr.io/kubernetes-e2e-test-images/volume/gluster:1.0 gcr.io/kubernetes-e2e-test-images/volume/nfs:1.0 google/cadvisor:latest k8s.gcr.io/busybox@sha256:4bdd623e848417d96127e16037743f0cd8b528c026e9175e22a84f639eca58ff k8s.gcr.io/e2e-test-images/agnhost:2.26 k8s.gcr.io/node-problem-detector:v0.6.2 k8s.gcr.io/pause:3.2 k8s.gcr.io/stress:v1]
W0131 19:16:55.059790   29670 image_list.go:199] Failed to pull  as user "root", retrying in 1s (1 of 5): exit status 1
W0131 19:16:56.129032   29670 image_list.go:199] Failed to pull  as user "root", retrying in 1s (2 of 5): exit status 1
W0131 19:16:57.201035   29670 image_list.go:199] Failed to pull  as user "root", retrying in 1s (3 of 5): exit status 1
W0131 19:16:58.266600   29670 image_list.go:199] Failed to pull  as user "root", retrying in 1s (4 of 5): exit status 1
W0131 19:16:59.343881   29670 image_list.go:199] Failed to pull  as user "root", retrying in 1s (5 of 5): exit status 1
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wzshiming**
**What type of PR is this?**
/kind bug
/sig testing

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #98628

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: test/e2e_node/image_list.go [hunk 1]
```diff
@@ -76,6 +76,10 @@ func updateImageAllowList() {
 	// Images from extra envs
 	framework.ImagePrePullList.Insert(getNodeProblemDetectorImage())
 	framework.ImagePrePullList.Insert(getSRIOVDevicePluginImage())
+
+	// Delete the empty string,
```
Comment (BenTheElder) [inline test/e2e_node/image_list.go:80]: why is there an empty string to begin with?

Comment (wzshiming) [reply test/e2e_node/image_list.go:80]: https://github.com/kubernetes/kubernetes/blob/7e6ef0efb6555bbd3720ac97bf913e8093fe7289/test/e2e_node/image_list.go#L216-L232
It is returned here. 
In the case of a bad network, an empty string may be returned.

But this should not affect the main process.

Comment (BenTheElder) [reply test/e2e_node/image_list.go:80]: if we're OK accepting this error we should just not insert the empty string to begin with, imho.
trying to remove it afterwards feels like a bit of a hack.

Comment (wzshiming) [reply test/e2e_node/image_list.go]: Agree. Updated.

**Comment by BenTheElder** [comment]
This looks about right to me, but I'd love SIG node to weigh in on this one, node_e2e is pretty much their thing 🙃

File: test/e2e_node/image_list.go [hunk 2]
```diff
@@ -60,12 +60,19 @@ var NodePrePullImageList = sets.NewString(
 	imageutils.GetE2EImage(imageutils.Perl),
 	imageutils.GetE2EImage(imageutils.Nonewprivs),
 	imageutils.GetPauseImageName(),
-	getGPUDevicePluginImage(),
 	imageutils.GetE2EImage(imageutils.NodePerfNpbEp),
 	imageutils.GetE2EImage(imageutils.NodePerfNpbIs),
 	imageutils.GetE2EImage(imageutils.NodePerfTfWideDeep),
 )
 
+func init() {
```
Comment (ehashman) [inline test/e2e_node/image_list.go:68]: Sticking this in `init` doesn't seem right when there's nothing else in here... is there somewhere else we can put this logic? Why is this only an issue for the `getGPUDevicePluginImage` and `getSRIOVDevicePluginImage`?

**Review by ehashman — COMMENTED** [review]
/cc @SergeyKanzhelev

File: test/e2e_node/image_list.go (hunk 2 above)
Comment (wzshiming) [reply test/e2e_node/image_list.go:68]: `getGPUDevicePluginImage` and `getSRIOVDevicePluginImage` in a bad network, may return empty string, causes subsequent pull image faill.or delete empty strings when using?

Comment (ehashman) [reply test/e2e_node/image_list.go:68]: Sure but there's also `getNodeProblemDetectorImage` which you're not changing. It just seems weird to me that we're managing this image in an `init()` and the other two images in a code block. If we are dynamically setting some images I think they should all be grouped together, as is this is very hard to follow.

Comment (BenTheElder) [reply test/e2e_node/image_list.go:68]: If we follow the pattern from the other calls then this has to insert _some_ value into the list, we could do that and put back `.Delete("")` on the end, but otherwise it's going to look fairly similar to this to accommodate maybe inserting maybe not?

IMHO this list should not be a global and should be initialized during main, so e.g. logging can be initialized first, but that's a larger change.

**Comment by SergeyKanzhelev** [comment]
@wzshiming please remove the release note, this is test code fix that doesn't need a release note.

/release-note-none
/priority backlog
/triage accepted
/test pull-kubernetes-unit 

This is currently waiting on author - please move the code to the place where other dynamic images are added.

**Comment by BenTheElder** [comment]
fixed the release note, missed that this also still needs code changes.

**Comment by wzshiming** [comment]
> fixed the release note, missed that this also still needs code changes.

I am not sure how to change this PR, whether to delete the empty ones after adding all of them, or to filter before adding?

**Comment by SergeyKanzhelev** [comment]
> > fixed the release note, missed that this also still needs code changes.
> 
> I am not sure how to change this PR, whether to delete the empty ones after adding all of them, or to filter before adding?

I think the feedback is to not do it in `Init` and move the logic out of `Init` to the same place where other dynamic images are added. Otherwise it looks OK

**Review by ehashman — COMMENTED** [review]
/lgtm
